test/parallel_api/ranges/std_ranges_memory.pass.cpp

```diff
@@ -0,0 +1,154 @@
+        };
+
+    test_memory_algo<Elem, -1>{}.run_host(dpl_ranges::uninitialized_fill, uninitialized_fill_checker, 2);
+
+    auto uninitialized_copy_move_checker =
+        [](const auto& res, auto&& r_in, auto&& r_out) {
+            using InRange = std::remove_cvref_t<decltype(r_in)>;
+            using OutRange = std::remove_cvref_t<decltype(r_out)>;
+
+            using Size = std::common_type_t<std::ranges::range_size_t<InRange>, std::ranges::range_size_t<OutRange>>;
+            const Size sz = std::ranges::min((Size)std::ranges::size(r_in), (Size)std::ranges::size(r_out));
+
+            const bool bres1 = (res.in == std::ranges::borrowed_iterator_t<InRange>(std::ranges::begin(r_in) + sz)
+                && res.out == std::ranges::borrowed_iterator_t<OutRange>(std::ranges::begin(r_out) + sz));
+
+            const bool bres2 = std::ranges::all_of(r_out, [](const auto& v) { return v.val1 == -1;})
+                && std::ranges::equal(std::ranges::take_view(r_in, sz), std::ranges::take_view(r_out, sz),
+                       [](const auto& v1, const auto& v2) { return v1.val2 == v2.val2;})
+                && std::ranges::all_of(std::ranges::drop_view(r_out, sz), [](const auto& v) { return v.val2 == -1;});
+
+            return std::pair<bool, bool>{bres1, bres2};
+        };
+
+    test_memory_algo<Elem, -1>{}.run_host(dpl_ranges::uninitialized_copy, uninitialized_copy_move_checker);
+    test_memory_algo<Elem_0, -1>{}.run_host(dpl_ranges::uninitialized_move, uninitialized_copy_move_checker);
+
+    auto destroy_checker =
+        [](const auto& res, const auto& r) {
+            using R = std::remove_cvref_t<decltype(r)>;
+            bool bres1 = (res == std::ranges::borrowed_iterator_t<R>(std::ranges::begin(r) + std::ranges::size(r)));
+            bool bres2 = std::ranges::all_of(r, [](const auto& v) { return v.val1 == -1;}) // -1 means no initialization
+                && std::ranges::all_of(r, [](const auto& v) { return v.val2 == 3;});
+
+            return std::pair<bool, bool>{bres1, bres2};
+        };
+
+    test_memory_algo<Elem_0, -1>{}.run_host(dpl_ranges::destroy, destroy_checker);
+
+#if TEST_DPCPP_BACKEND_PRESENT
+    test_memory_algo<Elem, -1>{}.run_device(dpl_ranges::uninitialized_default_construct, uninitialized_default_construct_checker);
+    test_memory_algo<Elem_0, -1>{}.run_device(dpl_ranges::uninitialized_value_construct, uninitialized_value_construct_checker);
+    test_memory_algo<Elem, -1>{}.run_device(dpl_ranges::uninitialized_fill, uninitialized_fill_checker, 2);
+
+    test_memory_algo<Elem, -1>{}.run_device(dpl_ranges::uninitialized_copy, uninitialized_copy_move_checker);
+    test_memory_algo<Elem_0, -1>{}.run_device(dpl_ranges::uninitialized_move, uninitialized_copy_move_checker);
+
+    test_memory_algo<Elem_0, -1>{}.run_device(dpl_ranges::destroy, destroy_checker);
+#endif //TEST_DPCPP_BACKEND_PRESENT
+
+#endif //_ENABLE_STD_RANGES_TESTING
+
+    return TestUtils::done(_ENABLE_STD_RANGES_TESTING);
```

Comment:
Let's add compilation checks for moved policies too like I did in the other ranges tests.

Reply:
It's inside `test_memory_algo`, no actions needed.